src/Application/Actions/Hooks/StripeUpdate.php

```diff
@@ -112,4 +116,72 @@ public function handleChargeSucceeded(Event $event): Response
 
         return $this->respondWithData($event->data->object);
     }
+
+    public function handlePayoutPaid(Event $event)
+    {
+        $stripeClient = new \Stripe\StripeClient($this->apiKey);
+
+        $payoutId = $event->data->object->id;
+
+        $this->logger->info('Getting all charges related to payout Id: ' . $payoutId);
+
+        $hasMore = true;
+        $lastBalanceTransactionId = null;
+        $paidChargeIds = [];
+        $attributes = [
+            'limit' => 100,
+            'payout' => $payoutId,
+            'type' => 'charge',
+        ];
+
+        while ($hasMore) {
+            $balanceTransactions = $stripeClient->balanceTransactions->all($attributes);
+
+            foreach ($balanceTransactions->data as $balanceTransaction) {
+                $paidChargeIds[] = $balanceTransaction->source;
+                $lastBalanceTransactionId = $balanceTransaction->id;
+            }
+
+            $hasMore = $balanceTransactions->has_more;
+
+            // We get a Stripe exception if we start this with a null or empty value,
```

Comment:
This doesn't make sense – if `$hasMore` is false then no more iterations of the `while` loop occur.

Reply:
@NoelLH the exception is encountered on the first call, where by default `$lastBalanceTransactionId` is set to null

Reply:
@tbg-jasonfung in that case, we can just check for that ID being non-null? Then it's clear why we're doing this and it's less likely to get removed or refactored incorrectly.